src/IP.cc

```diff
@@ -536,7 +536,8 @@ IPv6_Hdr_Chain::~IPv6_Hdr_Chain()
 	delete finalDst;
 	}
 
-void IPv6_Hdr_Chain::Init(const struct ip6_hdr* ip6, int total_len, bool set_next, uint16_t next)
+void IPv6_Hdr_Chain::Init(const struct ip6_hdr* ip6, uint64_t total_len, bool set_next,
```

Comment:
Only wonder if a smaller type or just an `unsigned int` would do? We hopefully never actually need the 64bit here and having it reads a bit like it could be.

But this isn't a well founded argument to change it.

Reply:
The warning being fixed here is this one:

```
..\src\Frag.cc(306): note: see reference to function template instantiation 'std::shared_ptr<zeek::IP_Hdr> std::make_shared<zeek::IP_Hdr,ip6_hdr*&,bool,uint64_t&,const zeek::IPv6_Hdr_Chain*&,bool>(ip6_hdr *&,bool &&,uint64_t &,const zeek::IPv6_Hdr_Chain *&,bool &&)' being compiled
```

The length value being passed here is built by Frag.cc as:

```
	// We have it all.  Compute the expected size of the fragment.
	uint64_t n = proto_hdr_len + frag_size;
```

`frag_size` is a `uint64_t`. Basically, it'd end up being a smaller change to make the argument take a `uint64_t`, even if it was a bit oversized, than to try to unwind changing the size of the values passed around in the fragment handler.